ITL-3 Implement edit and add state for todo card

BiroAron opened the pull request.

BiroAron committed:
ITL-7 Create the header component

BiroAron committed:
ITL-8 Add the title

BiroAron committed:
ITL-9 Add an add button, include a icon library or include the icon from the design

BiroAron committed:
ITL-10 Create the event that triggers the todo creation functionality

BiroAron committed:
Replaced plain CSS with Tailwind

BiroAron committed:
Made all the changes regarding to the comments

BiroAron committed:
Made the changes regarding to class names and reactive variables

BiroAron committed:
Removed all unnecessary code and comments

BiroAron committed:
Removed unnecessary id-s and redundant script

BiroAron committed:
Merge branch 'feature/ITL-1-implement-page-header' into develop

BiroAron committed:
ITL 11 Create the Placeholder img when the user is in the home page and no todos added to todo list

BiroAron committed:
ITL 12 Mobile friendly looking

BiroAron committed:
ITL 13  make the todo list look

BiroAron committed:
Made changes regarding the comments

-removed all of the conditional operators from the template tags
-removed unnecessary complicated code
-renamed non-descriptive variable names
-created a colorMap

BiroAron committed:
updatePriorityClass is now a variable instead of a function

BiroAron committed:
Merge branch 'feature/ITL-2-implement-todo-list' into develop

BiroAron committed:
ITL14 Subtask : Create an empty to-do item

BiroAron committed:
ITL 15Subtask 2: Display the new to-do item at the top of the list

BiroAron committed:
ITL 16Subtask 3: Set proper placeholder text for the new to-do item

BiroAron committed:
Created the requested file system

BiroAron committed:
ITL 17 Subtask 4 :After tapping the task, Card will convert it into edit mode

BiroAron committed:
ITL 19  Subtask 6 : Make the design match with figma

BiroAron committed:
ITL 20 Subtask 7: Pop-up confirmation

BiroAron committed:
Cleaned up the code

gerebdavid commented on src/components/todo/DeleteConfirmationPopup.vue:
```diff
@@ -0,0 +1,43 @@
+        </button>
+        <button
+          class="bg-red-500 text-white px-4 ml-1 py-2 rounded"
+          @click="deleItem(index)"
```
chnage this function name to deleteItem

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,160 @@
+          <input
+            type="text"
+            placeholder="Title"
+            v-model="props.todo.title"
```
move v-model to top

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,160 @@
+              </li>
+              <li>
+                <p
+                  class="block px-3 text-gray-80"
```
Change text-gray-80 to text-gray-800

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,160 @@
+          class="w-full p-2 mt-1 flex placeholder-gray-500 font-semibold focus:outline-none"
+          type="text"
+          placeholder="Description"
+          v-model="props.todo.text"
```
move v-model to top

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,160 @@
+        </button>
+        <DeleteConfirmationPopup
+          :index="index"
+          @close-popup="changePopupState"
```
first attributes, props and then events, emits

gerebdavid commented on src/components/todo/ToDoItem.vue:
```diff
@@ -0,0 +1,114 @@
+    :index="index"
+    @toggle-edit-visibility="toggleEditVisibility"
+    @delete-item="deleteItem(index)"
+    :class="getEditModeVisibility"
```
the order is wrong

gerebdavid commented on src/components/todo/ToDoItem.vue:
```diff
@@ -0,0 +1,114 @@
+  (e: 'deleteItem', index: number): void
+}>()
+
+const editingTodoVisibility = ref(false)
```
rename this to isEditingTodoVisible

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,160 @@
+}
+
+function updatePriority(priority: TodoPriority) {
+  props.todo.priority = priority
```
Please never change props values in the child components, always emit an event

BiroAron committed:
Made changes regarding the comments

- removed modification of props in child components

- component API orders corrected

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,164 @@
+      <div class="flex justify-between">
+        <div>
+          <input
+            v-model="props.todo.title"
```
Never use a v-model for a prop or a prop's attribute

gerebdavid commented on src/components/todo/ToDoForm.vue:
```diff
@@ -0,0 +1,164 @@
+      </div>
+      <div>
+        <textarea
+          v-model="props.todo.text"
```
same here

BiroAron committed:
Removed all reactive usage of props in child elements

BiroAron committed:
Divided the ToDoForm.vue into five subcomponents

kincsokiss commented on src/components/todo/ToDoItem.vue:
```diff
@@ -0,0 +1,127 @@
+  isEditingTodoVisible.value = !isEditingTodoVisible.value
+}
+
+function getChecIconVisisbility(todo: Todo) {
```
There is a misspelling issue in `getChecIconVisisbility ` -> `getChecIconVisibility `

kincsokiss commented on src/components/todo/ToDoPriority.vue:
```diff
@@ -0,0 +1,109 @@
+        <li>
+          <p
+            class="block px-3 text-black cursor-pointer"
+            @click="updatePriority('Low', index)"
```
Please create const variable for the priority and set the  `Low`, 'Medium' and `High` there. In this case use can use the variable to set the value of the priority instead of writing down the same string more times.
